.github/workflows/reusable_receptorctl_tests.yml

```
@@ -0,0 +1,53 @@
+---
+name: receptorctl
+
+"on":
+  workflow_call:
+
+jobs:
+  receptorctl:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version:
+        # NOTE: The highest and the lowest versions come
+        # NOTE: first as their statuses are most likely to
+        # NOTE: signal problems early:
+        - 3.11
+        - 3.8
+        - >-
+          3.10
+        - 3.9
+
+    env:
+      NOXSESSION: tests-${{ matrix.python-version }}
+
+    name: >-  # can't use `env` in this context:
+      Run `receptorctl` tests-${{ matrix.python-version }} session
+
+    steps:
+    - name: Download the `receptor` binary
+      uses: actions/download-artifact@v4
+      with:
+        name: receptor
+        path: /usr/local/bin/
+
+    - name: Set executable bit on the `receptor` binary
+      run: sudo chmod a+x /usr/local/bin/receptor
+
+    - name: Set up nox
+      uses: wntrblm/nox@2024.04.15
+      with:
+        python-versions: ${{ matrix.python-version }}
+
+    - name: Check out the source code from Git
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        sparse-checkout: receptorctl
+
+    - name: Run `receptorctl` nox ${{ env.NOXSESSION }} session
+      run: |
+        nox
+      working-directory: ./receptorctl
```

Let's separate provisioning from testing:
```suggestion
    - name: Provision nox environment for ${{ env.NOXSESSION }}
      run: nox --install-only
      working-directory: ./receptorctl
    - name: Run `receptorctl` nox ${{ env.NOXSESSION }} session
      run: nox
      working-directory: ./receptorctl
```

---

@webknjaz Not sure why we need to do this. And it looks like the sessions are failing now during provisioning.

---

It's needed to visualize failures of provisioning vs. failures of actual tests more clearly. I've suggested a fix already: https://github.com/ansible/receptor/pull/1003#pullrequestreview-2009328302.